src/kps/business/RouteCalculator.java

```diff
@@ -10,12 +10,13 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class RouteCalculator {
     private List<Route> routes;
 
     public RouteCalculator(Simulation simulation) {
-        this.routes = simulation.getRoutes();
+        this.routes = simulation.getUniqueRoutes().stream().filter(r -> !r.isDiscontinued()).collect(Collectors.toList());
```

Comment:
Ok. As far as I can see getUniqueRoutes is correct. When you say there are 3 routes from Wellington and Sydney, there is actually only 1 but there are 2 updates to that route. So getUniqueRoutes returns only the most recent route, which is discontinued